kubeflow/profiles/prototypes/README.md

```diff
@@ -0,0 +1,97 @@
+## Overview
+
+### Goals
+
+- Provide an easy way for data-scientists to create protected namespaces where 
+notebooks, jobs, and other components can be deployed into this namespace.
+
+- Use native kubernetes RBAC rules to isolate this namespace to a particular user or service account.
+
+- Separate infra components from user components where intra components reside in a shared namespace and 
+user components reside in protected namespaces.
+
+- Enable a forward path to include proposed [Security Profiles](https://github.com/kubernetes/community/blob/a8cb2060dc621664c86b185c7426367994b181b5/keps/draft-20180418-security-profile.md) 	
+
+
+## Design
+The Profiles ksonnet component provides a way for users and service accounts to run kubeflow components within isolated , protected namespaces. 3 CRDs are defined:
+
+- Profile
+- Target
+- Permissions
+
+The Profile resource contains a template section where a namespace and owner are specified. The Profile resource is created within the shared namespace. An example is:
+
+```
+apiVersion: kubeflow.org/v1alpha1
+kind: Profile
+metadata:
+  name: gan-alice
+  namespace: kubeflow
+spec:
+  template:
+    metadata:
+      name: gan
+    spec:
+      namespace: gan
+      owner: alice
+```
+
+The Target resource is created by the controller using the information in the Profile Resource. The Target resource is created within the shared namespace. The target contains a template where the name of the namespace and the permissions are specified. An example is:
```

Comment:
So does the user create the Target or is the Target automatically created after the user creates the profile?

Reply:
The latter, the controller watches for Profile and then creates a Target.
```
- profiles-controller 
  - watches for Profile
  - creates Target
```
```
- targets-controller
  - watches for Target
  - creates Namespace, Permission
```
```
- permissions-controller
  - watches for Permission
  - creates Role, RoleBinding
```